os.go

```diff
@@ -189,42 +190,39 @@ func isHidden(f os.FileInfo, path string, hiddenfiles []string) bool {
 	hidden := false
 	for _, pattern := range hiddenfiles {
 		matched := matchPattern(strings.TrimPrefix(pattern, "!"), f.Name(), path)
-		if strings.HasPrefix(pattern, "!") && matched {
-			hidden = false
-		} else if matched {
-			hidden = true
+		if !matched {
+			continue
 		}
+		hidden = !strings.HasPrefix(pattern, "!")
```

Comment:
I think you can simplify this even further:

```go
if matchPattern(strings.TrimPrefix(pattern, "!"), f.Name(), path) {
	hidden = !strings.HasPrefix(pattern, "!")
}
```

Reply:
That's even nicer, thanks!